site/lib/mergeHoldings.test.ts

```diff
@@ -0,0 +1,406 @@
+      token: "NCT",
+      tokenAmount: "1",
+      carbonValue: "1",
+      change: 1,
+    },
+    {
+      id: "0x838cd36c38e64c166b2c5130d15a8acff1e09d44-NCT-1674777600",
+      timestamp: "10",
+      token: "NCT",
+      tokenAmount: "2",
+      carbonValue: "2",
+      change: 1,
+    },
+    {
+      id: "0x838cd36c38e64c166b2c5130d15a8acff1e09d44-NCT-1674777600",
+      timestamp: "12",
+      token: "NCT",
+      tokenAmount: "3",
+      carbonValue: "3",
+      change: 1,
+    },
+    {
+      id: "0x838cd36c38e64c166b2c5130d15a8acff1e09d44-NCT-1674777600",
+      timestamp: "14",
+      token: "NCT",
+      tokenAmount: "4",
+      carbonValue: "4",
+      change: 1,
+    },
+  ],
+];
+
+const expectedOutput = {
+  BCT: [
+    {
+      carbonValue: "1",
+      id: "0xdeb8c24ad9640d62334da6f30c191e9cfd180e5b-BCT-1674777600",
+      timestamp: "1",
+      token: "BCT",
+      tokenAmount: "1",
+    },
+    {
+      carbonValue: "2",
+      id: "0xdeb8c24ad9640d62334da6f30c191e9cfd180e5b-BCT-1674777600",
+      timestamp: "3",
+      token: "BCT",
+      tokenAmount: "2",
+    },
+    {
+      carbonValue: "4",
+      id: "0x838cd36c38e64c166b2c5130d15a8acff1e09d44-BCT-1674777600",
+      timestamp: "4",
+      token: "BCT",
+      tokenAmount: "4",
+    },
+    {
+      carbonValue: "5",
+      id: "0xdeb8c24ad9640d62334da6f30c191e9cfd180e5b-BCT-1674777600",
+      timestamp: "5",
+      token: "BCT",
+      tokenAmount: "5",
+    },
+    {
+      carbonValue: "6",
+      id: "0x838cd36c38e64c166b2c5130d15a8acff1e09d44-BCT-1674777600",
+      timestamp: "6",
+      token: "BCT",
+      tokenAmount: "6",
+    },
+  ],
+  NCT: [
+    {
+      carbonValue: "1",
+      id: "0xdeb8c24ad9640d62334da6f30c191e9cfd180e5b-NCT-1674777600",
+      timestamp: "7",
+      token: "NCT",
+      tokenAmount: "1",
+    },
+    {
+      carbonValue: "2",
+      id: "0x838cd36c38e64c166b2c5130d15a8acff1e09d44-NCT-1674777600",
+      timestamp: "8",
+      token: "NCT",
+      tokenAmount: "2",
+    },
+    {
+      carbonValue: "3",
+      id: "0xdeb8c24ad9640d62334da6f30c191e9cfd180e5b-NCT-1674777600",
+      timestamp: "9",
+      token: "NCT",
+      tokenAmount: "3",
+    },
+    {
+      carbonValue: "4",
+      id: "0x838cd36c38e64c166b2c5130d15a8acff1e09d44-NCT-1674777600",
+      timestamp: "10",
+      token: "NCT",
+      tokenAmount: "4",
+    },
+    {
+      carbonValue: "5",
+      id: "0xdeb8c24ad9640d62334da6f30c191e9cfd180e5b-NCT-1674777600",
+      timestamp: "11",
+      token: "NCT",
+      tokenAmount: "5",
+    },
+    {
+      carbonValue: "6",
+      id: "0x838cd36c38e64c166b2c5130d15a8acff1e09d44-NCT-1674777600",
+      timestamp: "12",
+      token: "NCT",
+      tokenAmount: "6",
+    },
+    {
+      carbonValue: "7",
+      id: "0x838cd36c38e64c166b2c5130d15a8acff1e09d44-NCT-1674777600",
+      timestamp: "14",
+      token: "NCT",
+      tokenAmount: "7",
+    },
+  ],
+};
+
+export const holdingsTest = test("setup", () => {
```

Comment:
I've always done it with the `describe` and `test` pattern like this:

1. top level `describe` for each utility
2. inside there, multiple `tests`, one test for each of the expected behaviors/outcomes/edge cases
3. extra descriptive test names and comments so that newcomers can look at a test and understand all the requirements

```js
describe('groupHoldings', () => {

  test('Groups 2 holdings arrays of BCT and NCT', () => {
    expect(groupHoldings(holdingsInput)).toStrictEqual(groupedHoldings);
  });
  
  // test('other edge case', ...
});

describe('calculateChange', () => {

  test('Determine the balance change value for each entry', () => {
     expect(calculateChange(groupedHoldings)).toStrictEqual(
      calculatedGroupedHoldings
    );
  });
});

// etc
```

Reply:
Oh, right! Its been too long since I wrote a test lol